test/e2e/apps/statefulset.go

```diff
@@ -510,6 +510,152 @@ var _ = SIGDescribe("StatefulSet", func() {
+					currentRevision))
+			}
+
+			ginkgo.By("ADDING FINALIZER to pod0")
+			pod0name := getStatefulSetPodNameAtIndex(0, ss)
+			pod0, err := c.CoreV1().Pods(ns).Get(ctx, pod0name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+			pod0.Finalizers = []string{"example.com/myfinalizer"}
+			pod0, err = c.CoreV1().Pods(ss.Namespace).Update(ctx, pod0, metav1.UpdateOptions{})
+			framework.ExpectNoError(err)
+			pods.Items[0] = *pod0
+			defer func() {
+				err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+					ginkgo.By("REMOVING FINALIZER to POD0 - trying")
+					pod0, err := c.CoreV1().Pods(ss.Namespace).Get(ctx, pod0.Name, metav1.GetOptions{})
+					if err != nil {
+						ginkgo.By(fmt.Sprintf("Failed retrying to set the phase on pod0: %v", err))
+						return nil
+					}
+					pod0.Finalizers = []string{}
+					_, err = c.CoreV1().Pods(ss.Namespace).Update(ctx, pod0, metav1.UpdateOptions{})
+					if err == nil {
+						ginkgo.By("REMOVING FINALIZER to POD0 - success")
+						pods = e2estatefulset.GetPodList(ctx, c, ss)
+						for i := range pods.Items {
+							if i < int(*ss.Spec.UpdateStrategy.RollingUpdate.Partition) {
+								gomega.Expect(pods.Items[i].Spec.Containers[0].Image).To(gomega.Equal(oldImage), fmt.Sprintf("Pod %s/%s has image %s not equal to current image %s",
+									pods.Items[i].Namespace,
+									pods.Items[i].Name,
+									pods.Items[i].Spec.Containers[0].Image,
+									oldImage))
+								gomega.Expect(pods.Items[i].Labels[appsv1.StatefulSetRevisionLabel]).To(gomega.Equal(currentRevision), fmt.Sprintf("Pod %s/%s has revision %s not equal to current revision %s",
+									pods.Items[i].Namespace,
+									pods.Items[i].Name,
+									pods.Items[i].Labels[appsv1.StatefulSetRevisionLabel],
+									currentRevision))
+							} else {
+								gomega.Expect(pods.Items[i].Spec.Containers[0].Image).To(gomega.Equal(newImage), fmt.Sprintf("Pod %s/%s has image %s not equal to new image  %s",
+									pods.Items[i].Namespace,
+									pods.Items[i].Name,
+									pods.Items[i].Spec.Containers[0].Image,
+									newImage))
+								gomega.Expect(pods.Items[i].Labels[appsv1.StatefulSetRevisionLabel]).To(gomega.Equal(updateRevision), fmt.Sprintf("Pod %s/%s has revision %s not equal to new revision %s",
+									pods.Items[i].Namespace,
+									pods.Items[i].Name,
+									pods.Items[i].Labels[appsv1.StatefulSetRevisionLabel],
+									updateRevision))
+							}
+						}
+					}
+					return err
+				})
+				framework.ExpectNoError(err)
+			}()
+			// Delete all of the old pod.
+			deleteStatefulPodAtIndex(ctx, c, 0, ss)
+			ss = getStatefulSet(ctx, c, ss.Namespace, ss.Name)
+			currentRevision, updateRevision = ss.Status.CurrentRevision, ss.Status.UpdateRevision
+			gomega.Expect(currentRevision).ToNot(gomega.Equal(updateRevision), "Current revision should not equal update revision during rolling update")
+
+			pods = e2estatefulset.GetPodList(ctx, c, ss)
```

Comment:
ginkgo.By("Verify pod images")

Reply:
changed it with a more explanatory sentence.